libs/model/src/bot/CreateContest.command.ts

```diff
@@ -0,0 +1,91 @@
+import { ServerError, type Command } from '@hicommonwealth/core';
+import { commonProtocol as cp } from '@hicommonwealth/evm-protocols';
+import { config } from '@hicommonwealth/model';
+import * as schemas from '@hicommonwealth/schemas';
+import { buildFarcasterContestFrameUrl } from '@hicommonwealth/shared';
+import { models } from '../database';
+import { mustExist } from '../middleware/guards';
+import { TokenAttributes } from '../services';
+import { contractHelpers } from '../services/commonProtocol';
+import { deployERC20Contest } from '../services/commonProtocol/contestHelper';
+import { parseBotCommand } from '../services/openai/parseBotCommand';
+
+export function CreateContest(): Command<typeof schemas.CreateBotContest> {
+  return {
+    ...schemas.CreateBotContest,
+    auth: [],
+    body: async ({ payload }) => {
+      const { prompt, chain_id } = payload;
+      const contestMetadata = await parseBotCommand(prompt);
+
+      const namespaceFactory =
+        cp.factoryContracts[chain_id as cp.ValidChains].factory;
+      const botNamespace = config.BOT.CONTEST_BOT_NAMESPACE;
+
+      if (!botNamespace || botNamespace === '') {
+        new ServerError('bot not enabled on given chain');
+      }
+
+      const community = await models.Community.scope('withPrivateData').findOne(
+        {
+          where: {
+            namespace: botNamespace as string,
+          },
+        },
+      );
+
+      if (!community) {
```

Comment:
you can also use the mustExist guard to throw InvalidState when community is falsy, type-narrow the variable to non-nullish